components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php

```diff
@@ -0,0 +1,417 @@
+    private ilAuthLogoutPageEditorSettings $settings;
+    private ?ilSetting $logoutSettings = null;
+    protected \ILIAS\Style\Content\Object\ObjectFacade $content_style_domain;
+
+    //variables from requests
+    private ?string $redirect_source = null;
+    private ?int $key = null;
+    private array $visible_languages = [];
+    private array $languages = [];
+
+    public function __construct(int $a_ref_id)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->tpl = $DIC->ui()->mainTemplate();
+        $this->tabs = $DIC->tabs();
+        $this->toolbar = $DIC->toolbar();
+        $this->rbacsystem = $DIC->rbac()->system();
+        $this->setting = $DIC->settings();
+        $this->ilErr = $DIC['ilErr'];
+
+        $this->http = $DIC->http();
+        $this->refinery = $DIC->refinery();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+
+        $this->lng = $DIC['lng'];
+
+        $this->lng->loadLanguageModule('auth');
+        $this->ref_id = $a_ref_id;
+
+        $this->settings = ilAuthLogoutPageEditorSettings::getInstance();
+        $this->content_style_domain = $DIC->contentStyle()
+                                          ->domain()
+                                          ->styleForRefId($a_ref_id);
+
+        $query_wrapper = $DIC->http()->wrapper()->query();
+        $post_wrapper = $DIC->http()->wrapper()->post();
+        $is_post_request = $DIC->http()->request()->getMethod() === "POST";
+        $refinery = $DIC->refinery();
+
+        if ($query_wrapper->has("redirectSource")) {
+            $this->redirect_source = $query_wrapper->retrieve("redirectSource", $refinery->kindlyTo()->string());
+        }
+
+        if ($post_wrapper->has("key")) {
+            $this->key = $post_wrapper->retrieve("key", $refinery->kindlyTo()->int());
+        } elseif ($query_wrapper->has("key")) {
+            $this->key = $query_wrapper->retrieve("key", $refinery->kindlyTo()->int());
+        }
+
+        if ($is_post_request) {
+            if ($post_wrapper->has("visible_languages")) {
+                $this->visible_languages = $post_wrapper->retrieve(
+                    "visible_languages", $refinery->kindlyTo()->listOf($refinery->kindlyTo()->string())
+                );
+            }
+
+            if ($post_wrapper->has("languages")) {
+                $this->languages = $post_wrapper->retrieve(
+                    "languages", $refinery->kindlyTo()->listOf($refinery->kindlyTo()->string())
+                );
+            }
+        }
+    }
+
+    public function getSettings(): ilAuthLogoutPageEditorSettings
+    {
+        return $this->settings;
+    }
+
+    public function getRefId(): int
+    {
+        return $this->ref_id;
+    }
+
+    /**
+     * ilCtrl execute command
+     */
+    public function executeCommand(): void
+    {
+        switch ($this->ctrl->getNextClass($this)) {
+            case 'illogoutpagegui':
```

Comment:
Please use the class-string and normalize this (`strtolower`, as well as the string of the switch statement).

Reply:
done